lib/phx_tools_web/live/phx_tools_live/index.ex

```diff
@@ -2,31 +2,56 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
   use PhxToolsWeb, :live_view
 
   alias PhxToolsWeb.Endpoint
-  alias PhxToolsWeb.InstructionComponents
-  alias PhxToolsWeb.PhxToolsLive.LandingComponent
+  alias PhxToolsWeb.PhxToolsComponents
 
   @impl Phoenix.LiveView
   def mount(_params, session, socket) do
     {:ok,
      socket
      |> assign(:seo_attributes, %{})
-     |> assign_os(session)}
+     |> assign_os_and_source_code_url(session)}
   end
 
-  defp assign_os(socket, %{"operating_system" => operating_system}),
-    do: assign(socket, :operating_system, operating_system)
+  defp assign_os_and_source_code_url(socket, %{"operating_system" => operating_system}) do
+    socket
+    |> assign(:operating_system, operating_system)
+    |> assign(:source_code_url, source_code_url(operating_system))
+  end
 
   @impl Phoenix.LiveView
-  def handle_params(_params, _uri, socket),
-    do: {:noreply, apply_action(socket, socket.assigns.live_action)}
+  def handle_params(_params, _uri, socket) do
+    apply_action(socket, socket.assigns.live_action)
+  end
+
+  defp apply_action(socket, :index) do
+    case socket.assigns.operating_system do
+      "Linux" -> {:noreply, assign(socket, :live_action, :linux)}
+      "macOS" -> {:noreply, assign(socket, :live_action, :macOS)}
+      _other -> {:noreply, socket}
+    end
+  end
+
+  defp apply_action(socket, action) do
+    {:noreply,
+     socket
+     |> assign(seo_attributes: %{url: Endpoint.url() <> "/#{action}"})
+     |> assign(:source_code_url, source_code_url(action))}
+  end
 
-  defp apply_action(socket, :linux) do
-    assign(socket, seo_attributes: %{url: url(~p"/linux")})
+  defp installation_command(live_action) do
+    "/bin/bash -c \"$(curl -fsSL #{Endpoint.url() <> "/#{capitalize_os_name_first_letter("#{live_action}")}.sh"})\""
   end
 
-  defp apply_action(socket, :macOS) do
-    assign(socket, seo_attributes: %{url: url(~p"/macOS")})
+  defp source_code_url(os) do
+    "https://github.com/optimumBA/phx.tools/blob/main/priv/static/#{capitalize_os_name_first_letter("#{os}")}.sh"
   end
```

Comment:
This may have been possible if I was not using the capitalize_os_name_first_letter (now get_os_from_live_action) elsewhere where the .sh extension is not needed.